wire-ios-sync-engine/Source/Synchronization/SyncAgent.swift

```diff
@@ -207,6 +211,35 @@ final class SyncAgent: NSObject, SyncAgentProtocol {
 
 }
 
+// MARK: - MLS sync delegate
+
+extension SyncAgent: MLSSyncDelegate {
+
+    func recoverWithIncrementalSync() async throws {
+        WireLogger.sync.info("performing recovery incremental sync")
+
+        if isSyncV2Enabled {
+            // Recovery means to restart any existing sync.
+            await suspend()
+
+            do {
+                try await incrementalSyncTaskManager.performIfNeeded { [weak self] in
+                    guard let self else { return }
+                    delegate?.syncAgentDidStartIncrementalSync(self)
+                    incrementalSyncToken = try await incrementalSyncProvider.provideIncrementalSync().perform()
+                    delegate?.syncAgentDidFinishIncrementalSync(self, isRecovering: true)
+                }
```

Comment:
question: does it do two sync then ?

Reply:
no it should do either new sync or old sync. Where do you see the two syncs?

Reply:
I am confused by:
```
// Recovery means to restart any existing sync.
            await suspend()
 ```
 
 but i guess this stops any ongoing sync, abd start a new one right?

Reply:
correct, stop the existing one, start a new one.